### DISCUSSION TITLE
consume always most new message

### DISCUSSION THREAD

**Discussion by bisegni** [edenhill/librdkafka#3747] [Q&A]
there is a way for a consume to get always the last available message from Kafka? I mean this step:

1) poll for get the new message
2) with some time
3) poll for a newly most recent message discarding all received message sins first poll except the new on

I want only the new message at poll time ( I need it to monitor the most recent state and not for consume and process all message). Immagine to have under of ui that need only to monitor but only few consumer that process data in realtime

thanks for help

**Comment by edenhill** [comment]
Let's say you want to continously show the last 10 messages for a partition, in that case I would set `enable.partition.eof=true` and keep a rolling window of the last 5 messages, whenever you see a PARTITION_EOF error you show the last 5 msgs from your window.

**Reply by bisegni** [reply]
hi thanks you, I can I set parameters for take the 5 message window before each pool ?
thanks in advance
